src/commcare_cloud/ansible/deploy_hq.yml

```diff
@@ -9,6 +9,7 @@
     - pillowtop
     - proxy
     - django_manage
+  any_errors_fatal: true
```

Comment:
I'm surprised this is needed; I thought this was the default behavior in ansible. I thought if any task failed (unless it's specifically handled) the ansible run would end. But the documentation for `any_errors_fatal` matches what you're saying (https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html#aborting-on-the-first-error-any-errors-fatal), so now I'm confused as to what the default behavior is.

Reply:
The default behaviour is: "Ansible continues to execute tasks as long as there are hosts that have not yet failed" (https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html#setting-a-maximum-failure-percentage)

So if a task fails on a host then it will be ignored for the rest of the playbook but other hosts will continue to have tasks executed on them.

Reply:
Yes, from the top of the [page you linked](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html#error-handling-in-playbooks):

> When Ansible receives a non-zero return code from a command or a failure from a module, by default it stops executing on that host and continues on other hosts.

Although [loops appear to be an exception](https://github.com/dimagi/commcare-cloud/blob/5118f0aa33d13e020c86ce02ade273e2928e8f3c/src/commcare_cloud/ansible/roles/deploy_hq/tasks/staticfiles_collect.yml#L21-L26). Execution does not stop until the loop is exhausted.

I wish there was a way to set `any_errors_fatal: true` globally when running a playbook rather than adding it to every play, but I haven't found a way to do that.